scripts.py

```diff
@@ -995,6 +1004,340 @@ class OPDSForDistributorsReaperScript(OPDSImportScript):
+            required=True
+        )
+        RS = RightsStatus
+        rights_uris = ", ".join(RS.OPEN_ACCESS)
+        parser.add_argument(
+            '--rights-uri',
+            help=u"A URI explaining the rights status of the works being uploaded. Acceptable values: %s" % rights_uris,
+            required=True
+        )
+        parser.add_argument(
+            '--dry-run',
+            help=u"Show what would be imported, but don't actually do the import.",
+            action='store_true',
+        )
+        return parser
+
+    def do_run(self, cmd_args=None):
+        parser = self.arg_parser(self._db)
+        parsed = parser.parse_args(cmd_args)
+        collection_name = parsed.data_source_name
+        data_source_name = parsed.data_source_name
+        metadata_file = parsed.metadata_file
+        cover_directory = parsed.cover_directory
+        ebook_directory = parsed.ebook_directory
+        rights_uri = parsed.rights_uri
+        dry_run = parsed.dry_run
+        return self.run_with_arguments(
+            collection_name, data_source_name,
+            metadata_file, cover_directory,
+            ebook_directory, rights_uri, dry_run
+        )
+
+    def run_with_arguments(
+            self, collection_name, data_source_name,
+            metadata_file, cover_directory, ebook_directory, rights_uri,
+            dry_run
+    ):
+        if dry_run:
+            self.log.warn(
+                "This is a dry run. No files will be uploaded and nothing will change in the database."
+            )
+        collection, mirror = self.load_collection(
+            collection_name, data_source_name
+        )
+
+        if dry_run:
+            mirror = None
+
+        replacement_policy = ReplacementPolicy.from_license_source(self._db)
+        replacement_policy.mirror = mirror
+        metadata_records = self.load_metadata(metadata_file)
+        for metadata in metadata_records:
+            self.work_from_metadata(
+                collection, metadata, replacement_policy, cover_directory, 
+                ebook_directory, rights_uri
+            )
+            if not dry_run:
+                self._db.commit()
+
+    def load_collection(self, collection_name, data_source_name):
+        """Create or locate a Collection with the given name.
+
+        If the Collection needs to be created, it will be associated
+        with the given data source and (if configured) the site-wide
+        mirror configuration.
+
+        :param collection_name: Name of the Collection.
+        :param data_source_name: Associate this data source with
+            the Collection if it does not already have a data source.
+            A DataSource object will be created if necessary.
+
+        :return: A 2-tuple (Collection, MirrorUploader)
+        """
+        collection, is_new = Collection.by_name_and_protocol(
+            self._db, collection_name, ExternalIntegration.DIRECTORY_IMPORT
```

Comment:
I might go with something like "Manual import" as the protocol to leave open the possibility of adding individual books to this collection from the admin interface as well. I'm not sure we need to distinguish that it was a bulk directory import.

Reply:
Would "Manual intervention" or "Library staff" be appropriate? We already have constants for those.

Reply:
I think "Manual intervention" sounds more like a protocol than "Library staff".